core/src/Kokkos_View.hpp

```diff
@@ -540,13 +632,14 @@ class View : public ViewTraits<DataType, Properties...> {
 
  public:
   typedef ViewTraits<DataType, Properties...> traits;
+  typedef Kokkos::Impl::ViewTracker<View> view_tracker_type;
 
  private:
   typedef Kokkos::Impl::ViewMapping<traits, typename traits::specialize>
       map_type;
-  typedef Kokkos::Impl::SharedAllocationTracker track_type;
+  friend view_tracker_type;
```

Comment:
I don't think this is portable. Friending something dependent on a template parameter chokes in some compilers IIRC. I'd probably just friend the whole template.

Reply:
fixed